### PR TITLE
Gaussian blur approval

### PR DESCRIPTION
**Pull request by artu1999**
pull request with correct author email

### CODE REVIEW

**Comment by LukeWood** [comment]
Thanks for fixing!  Can review tomorrow!

File: keras_cv/layers/preprocessing/gaussian_blur.py [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+
+import tensorflow as tf
+
+
+@tf.keras.utils.register_keras_serializable(package="keras_cv")
+class GaussianBlur(tf.keras.__internal__.layers.BaseImageAugmentationLayer):
+    """Applies a Gaussian Blur with random sigma to an image.
+
+    Args:
+        kernel_size: int, 2 element tuple or 2 element list. x and y dimensions for
```
Comment (LukeWood) [inline keras_cv/layers/preprocessing/gaussian_blur.py:23]: Check out how we are handling sampling like this.  `sigma` could use this.  There is a nice utility to parse these ranges for you.  See https://github.com/keras-team/keras-cv/pull/238

This isn't merged yet - but you can rebase on the branch `factor` to use this.

File: keras_cv/layers/preprocessing/gaussian_blur.py [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+    def get_random_transformation(self, image=None, label=None, bounding_box=None):
+        sigma = self.get_sigma()
+        blur_v = GaussianBlur.get_kernel(sigma, self.y)
+        blur_h = GaussianBlur.get_kernel(sigma, self.x)
+        blur_v = tf.reshape(blur_v, [self.y, 1, 1, 1])
+        blur_h = tf.reshape(blur_h, [1, self.x, 1, 1])
+        return (blur_v, blur_h)
+
+    def get_sigma(self):
+        sigma = self._random_generator.random_uniform(
```
Comment (LukeWood) [inline keras_cv/layers/preprocessing/gaussian_blur.py:87]: Nicely done!

File: keras_cv/layers/preprocessing/gaussian_blur_test.py [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+        # greyscale
+        xs = tf.cast(
+            tf.ones((512, 512, 1)),
+            dtype=tf.float32,
+        )
+
+        xs = layer(xs)
+        self.assertEqual(xs.shape, [512, 512, 1])
+
+    def test_non_square_images(self):
```
Comment (LukeWood) [inline keras_cv/layers/preprocessing/gaussian_blur_test.py:55]: Lets also implement a numerical test.  In order to do this, you can pass `sigma=(0.3, 0.3)` and compute desired output values for a given image.  For this image, may I recommend:

```
0 0 0
0 1 0
0 0 0
```

Then you can check what the values are on for the 0s, and ensure the filter is correctly applied.  Looking really good!

**Review by LukeWood — CHANGES_REQUESTED** [review]

**Comment by LukeWood** [comment]
@artu1999 thanks for the PR!  This is looking really good!  I left some minor comments - biggest one is adding the new test case!

**Comment by artu1999** [comment]
@LukeWood Still getting problems with the CLA agreement... the email in my config is the one of the contract, not sure why this is happening

**Comment by LukeWood** [comment]
> @LukeWood Still getting problems with the CLA agreement... the email in my config is the one of the contract, not sure why this is happening

Can you accept it with your other email?  It is probably the author of one of your commits.

**Comment by artu1999** [comment]
> > @LukeWood Still getting problems with the CLA agreement... the email in my config is the one of the contract, not sure why this is happening
> 
> Can you accept it with your other email? It is probably the author of one of your commits.
 
I checked the CLA and it's the email github gave to my laptop, so I can't verify and sign the email. I was sure I changed it last time to the email in the CLA when I got the checks right, apologies for the mess. Could it help if I push from another branch and I double check before the PR that the email is set right?

**Comment by LukeWood** [comment]
> > > @LukeWood Still getting problems with the CLA agreement... the email in my config is the one of the contract, not sure why this is happening
> > 
> > 
> > Can you accept it with your other email? It is probably the author of one of your commits.
> 
> I checked the CLA and it's the email github gave to my laptop, so I can't verify and sign the email. I was sure I changed it last time to the email in the CLA when I got the checks right, apologies for the mess. Could it help if I push from another branch and I double check before the PR that the email is set right?

Feel free to push to another branch with a single commit authored by the right email!  No worries @artu1999 these things can be a pain!

**Pull request closed without merging**